tests/capture/transforms/test_mcm_execution.py

```diff
@@ -0,0 +1,299 @@
+        """Test that using qml.cond with non-MCM predicates works as expected."""
+
+        dev = qml.device("default.qubit", wires=5)
+
+        @DeferMeasurementsInterpreter(num_wires=5)
+        def f(x):
+            qml.Hadamard(0)
+            m0 = qml.measure(0)
+
+            @qml.cond(x > 2.5)
+            def cond_fn():
+                qml.RX(m0 * jnp.pi, 0)
+                # Final state |0>
+
+            @cond_fn.else_if(x > 1.5)
+            def _():
+                qml.PauliZ(0)
+                # Equal prob of |0> and |1>
+
+            @cond_fn.otherwise
+            def _():
+                qml.Hadamard(0)
+                m1 = qml.measure(0)
+                qml.RX(m1 * jnp.pi, 0)
+                qml.X(0)
+                # Final state |1>
+
+            cond_fn()
+
+            return qml.expval(qml.PauliZ(0))
+
+        jaxpr = jax.make_jaxpr(f)(0.5)
+
+        arg_true = 3.0
+        res = dev.eval_jaxpr(
+            jaxpr.jaxpr, jaxpr.consts, arg_true, execution_config=create_execution_config()
+        )
+        assert qml.math.allclose(res, 1)  # Final state |0>; <Z> = 1
+
+        arg_elif = 2.0
+        res = dev.eval_jaxpr(
+            jaxpr.jaxpr, jaxpr.consts, arg_elif, execution_config=create_execution_config()
+        )
+        assert qml.math.allclose(res, 0)  # Equal prob of |0>, |1>; <Z> = 1
+
+        arg_true = 1.0
+        res = dev.eval_jaxpr(
+            jaxpr.jaxpr, jaxpr.consts, arg_true, execution_config=create_execution_config()
+        )
+        assert qml.math.allclose(res, -1)  # Final state |1>, <Z> = -1
+
+    @pytest.mark.parametrize(
+        "mp_fn",
+        [qml.expval, qml.var, qml.probs],
+    )
+    def test_mcm_statistics(self, mp_fn):
+        """Test that collecting statistics on MCMs is handled correctly."""
+
+        dev = qml.device("default.qubit", wires=5)
+
+        def processing_fn(m1, m2):
+            return 2.5 * m1 - m2
+
+        def f():
+            qml.Hadamard(0)
+            m0 = qml.measure(0)
+            qml.Hadamard(0)
+            m1 = qml.measure(0)
+            qml.Hadamard(0)
+            m2 = qml.measure(0)
+
+            outs = (mp_fn(op=m0),)
+            if mp_fn is qml.probs:
+                outs += (mp_fn(op=[m0, m1, m2]),)
+            else:
+                outs += (mp_fn(op=processing_fn(m1, m2)),)
+
+            return outs
+
+        transformed_f = DeferMeasurementsInterpreter(num_wires=5)(f)
+
+        jaxpr = jax.make_jaxpr(transformed_f)()
+        res = dev.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, execution_config=create_execution_config())
+
+        with qml.capture.pause():
+            qnode_f = qml.QNode(f, dev, mcm_method="deferred")
+            expected = qnode_f()
+
+        for r, e in zip(res, expected, strict=True):
+            assert qml.math.allclose(r, e)
+
+
+# pylint: disable=too-few-public-methods
+class TestExecutionFiniteShots:
+    """Tests for executing circuits with finite shots."""
```

Comment:
Not really sure what else to test in this test class. I added some skeleton tests, but I don't even know if those tests make sense.